packages/aws-cdk/lib/api/deploy-stack.ts

```diff
@@ -478,16 +478,24 @@ class FullCloudFormationDeployment {
     const startTime = new Date();
 
     if (this.update) {
-      await this.cfn.updateStack({
-        StackName: this.stackName,
-        ClientRequestToken: `update${this.uuid}`,
-        ...this.commonPrepareOptions(),
-        ...this.commonExecuteOptions(),
-      }).promise();
-
-      const ret = await this.monitorDeployment(startTime, undefined);
       await this.updateTerminationProtection();
-      return ret;
+
+      try {
+        await this.cfn.updateStack({
+          StackName: this.stackName,
+          ClientRequestToken: `update${this.uuid}`,
+          ...this.commonPrepareOptions(),
+          ...this.commonExecuteOptions(),
+        }).promise();
+      } catch (err: any) {
+        if (err.message === 'No updates are to be performed.') {
+          debug('No updates are to be performed for stack %s', this.stackName);
+          return { noOp: true, outputs: this.cloudFormationStack.outputs, stackArn: this.cloudFormationStack.stackId };
+        }
+        throw err;
+      }
+
+      return this.monitorDeployment(startTime, undefined);
```

Comment:
Why was the `cfn.updateStack` call moved after the `updateTerminationProtection()` method?